src/globus_sdk/experimental/auth_requirements_error/_validators.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+import sys
+import typing as t
+
+from ._serializable import Serializable
+
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+S = t.TypeVar("S", bound=Serializable)
+
+
+class ValidationError(ValueError):
+    pass
+
+
+def str_(name: str, value: t.Any) -> str:
+    if not isinstance(value, str):
+        raise ValidationError(f"'{name}' must be a string")
+    return value
+
+
+def opt_str(name: str, value: t.Any) -> str | None:
+    if value is None:
+        return None
+    if not isinstance(value, str):
+        raise ValidationError(f"'{name}' must be a string")
+    return value
+
+
+def consent_required_literal(name: str, value: t.Any) -> Literal["ConsentRequired"]:
+    if not isinstance(value, str) or value != "ConsentRequired":
+        raise ValidationError(f"'{name}' must be the string 'ConsentRequired'")
+    return t.cast(Literal["ConsentRequired"], value)
```

Comment:
This feels to me a little bit like underlying details of specific variants are leaking into the validators, which makes this feel a little muddy. Not a hill I'm going to de on, and wouldn't want to spend more than a few minutes on it, but would prefer it if `ConsentRequired` was limited to the variants module.

Reply:
I don't have a strong stake in this being here vs there. Either way it's an implementation detail.

I'll happily move it to the end of `_variants.py`.

Reply:
Done! :smile: